# Import Modules

In [3]:
# Import required module
import pandas as pd
import time
from selenium import webdriver
  
# Import Select class
from selenium.webdriver.support.ui import Select

ModuleNotFoundError: ignored

# Parameters

In [ ]:
# TODO: Update section with paths


# Parsing enforcementtracker.com

## Parsing high level table

In [ ]:
driver = webdriver.Edge('../srcs/msedgedriver.exe')
driver.get('https://www.enforcementtracker.com/')
time.sleep(4)

# Select dropdown by name
x = driver.find_element_by_name('penalties_length')
drop = Select(x)
  
# Select by value
drop.select_by_value("100")
time.sleep(4)


In [ ]:
table = driver.find_element_by_xpath('//*[@id="penalties_wrapper"]')
table_html = table.get_attribute('innerHTML')


In [ ]:
df = pd.read_html(table_html)[0]
df.dropna(how='all', inplace=True)
df


Unnamed: 0      ETid  Country Date of Decision Fine [€]  \
1           NaN  ETid-914    SPAIN       2021-11-23    40000   
2           NaN  ETid-913    SPAIN       2021-11-23     3000   
3           NaN  ETid-912    SPAIN       2021-11-22     2000   
4           NaN  ETid-911    SPAIN       2021-11-22     3000   
5           NaN  ETid-910   FRANCE       2021-11-04   400000   
..          ...       ...      ...              ...      ...   
96          NaN  ETid-819    SPAIN       2021-08-25   120000   
97          NaN  ETid-818    SPAIN       2021-08-26     1000   
98          NaN  ETid-817   POLAND       2021-08-13     2200   
99          NaN  ETid-816  ROMANIA       2021-08-24     3000   
100         NaN  ETid-815    SPAIN       2021-08-23     2000   

                        Controller/Processor  \
1                       Vodafone España, SAU   
2                             FUENSANTA S.L.   
3                 ANIVERSALIA NETWORKS, S.L.   
4                                    Unknown   
5    Régie autonome des transports parisiens   
..                                       ...   
96     Banco Bilbao Vizcaya Argentaria, S.A.   
97                        Owners Association   
98    President of the Zgierz District Court   
99                            Actamedica SRL   
100                            Company owner   

                                           Quoted Art.  \
1                                      Art. 6 (1) GDPR   
2                                     Art. 58 (1) GDPR   
3                                         Art. 13 GDPR   
4                                         Art. 13 GDPR   
5    Art. 5 (1) c) GDPR, Art. 5 (1) e) GDPR, Art. 5...   
..                                                 ...   
96                                        Art. 32 GDPR   
97                                  Art. 5 (1) c) GDPR   
98   Art. 5 (1) f) GDPR, Art. 25 (1) GDPR, Art. 32 ...   
99        Art. 28 (1) GDPR, Art. 32 GDPR, Art. 33 GDPR   
100                                       Art. 13 GDPR   

                                                  Type     Source  
1         Insufficient legal basis for data processing       link  
2    Insufficient cooperation with supervisory auth...       link  
3    Insufficient fulfilment of information obligat...       link  
4    Insufficient fulfilment of information obligat...       link  
5    Non-compliance with general data processing pr...  link link  
..                                                 ...        ...  
96   Insufficient technical and organisational meas...       link  
97   Non-compliance with general data processing pr...       link  
98   Insufficient technical and organisational meas...       link  
99   Insufficient technical and organisational meas...       link  
100  Insufficient fulfilment of information obligat...       link  

[100 rows x 9 columns]

In [ ]:
# number of pages to parse
page_no = 10

for i in range(1,page_no):
    print(i)
    time.sleep(20)
    try:
        driver.find_element_by_id('penalties_next').click()
    except:
        print("Next Click failed. Trying it again.")
        time.sleep(20)
        driver.find_element_by_id('penalties_next').click()
    time.sleep(4)
    table = driver.find_element_by_xpath('//*[@id="penalties_wrapper"]')
    table_html = table.get_attribute('innerHTML')
    df2 = pd.read_html(table_html)[0]
    df2.dropna(how='all', inplace=True)
    df = df.append(df2)

1
Next Click failed. Trying it again.
2
3
4
5
6
7
8


KeyboardInterrupt: 

In [ ]:
df.drop(['Unnamed: 0', 'Source'], axis=1, inplace=True)

In [ ]:
df

ETid  Country Date of Decision Fine [€]  \
1   ETid-914    SPAIN       2021-11-23    40000   
2   ETid-913    SPAIN       2021-11-23     3000   
3   ETid-912    SPAIN       2021-11-22     2000   
4   ETid-911    SPAIN       2021-11-22     3000   
5   ETid-910   FRANCE       2021-11-04   400000   
..       ...      ...              ...      ...   
10    ETid-5  BELGIUM       2019-05-28   2000.0   
11    ETid-4  AUSTRIA       2018-12-20   2200.0   
12    ETid-3  AUSTRIA       2018-09-27    300.0   
13    ETid-2  AUSTRIA             2018   1800.0   
14    ETid-1  AUSTRIA       2018-12-09   4800.0   

                       Controller/Processor  \
1                      Vodafone España, SAU   
2                            FUENSANTA S.L.   
3                ANIVERSALIA NETWORKS, S.L.   
4                                   Unknown   
5   Régie autonome des transports parisiens   
..                                      ...   
10                                    Mayor   
11                           Private person   
12                        Private car owner   
13                         Kebab restaurant   
14                            Betting place   

                                          Quoted Art.  \
1                                     Art. 6 (1) GDPR   
2                                    Art. 58 (1) GDPR   
3                                        Art. 13 GDPR   
4                                        Art. 13 GDPR   
5   Art. 5 (1) c) GDPR, Art. 5 (1) e) GDPR, Art. 5...   
..                                                ...   
10                    Art. 5 (1) b) GDPR, Art. 6 GDPR   
11  Art. 5 (1) a) GDPR, Art. 5 (1) c) GDPR, Art. 6...   
12                    Art. 5 (1) a) GDPR, Art. 6 GDPR   
13            Art. 5 GDPR, Art. 13 GDPR, Art. 14 GDPR   
14                                       Art. 13 GDPR   

                                                 Type  
1        Insufficient legal basis for data processing  
2   Insufficient cooperation with supervisory auth...  
3   Insufficient fulfilment of information obligat...  
4   Insufficient fulfilment of information obligat...  
5   Non-compliance with general data processing pr...  
..                                                ...  
10       Insufficient legal basis for data processing  
11       Insufficient legal basis for data processing  
12       Insufficient legal basis for data processing  
13       Insufficient legal basis for data processing  
14  Insufficient fulfilment of information obligat...  

[914 rows x 7 columns]

In [ ]:
# Using pickle format to preserve formats
pd.to_pickle(df,"../data/external/gdpr_fines_enforcementtracker.pkl", protocol=2)

In [ ]:
driver.close()

In [ ]:
df.set_index('ETid', inplace=True)

In [ ]:
df

Country Date of Decision Fine [€]  \
ETid                                          
ETid-914    SPAIN       2021-11-23    40000   
ETid-913    SPAIN       2021-11-23     3000   
ETid-912    SPAIN       2021-11-22     2000   
ETid-911    SPAIN       2021-11-22     3000   
ETid-910   FRANCE       2021-11-04   400000   
...           ...              ...      ...   
ETid-5    BELGIUM       2019-05-28   2000.0   
ETid-4    AUSTRIA       2018-12-20   2200.0   
ETid-3    AUSTRIA       2018-09-27    300.0   
ETid-2    AUSTRIA             2018   1800.0   
ETid-1    AUSTRIA       2018-12-09   4800.0   

                             Controller/Processor  \
ETid                                                
ETid-914                     Vodafone España, SAU   
ETid-913                           FUENSANTA S.L.   
ETid-912               ANIVERSALIA NETWORKS, S.L.   
ETid-911                                  Unknown   
ETid-910  Régie autonome des transports parisiens   
...                                           ...   
ETid-5                                      Mayor   
ETid-4                             Private person   
ETid-3                          Private car owner   
ETid-2                           Kebab restaurant   
ETid-1                              Betting place   

                                                Quoted Art.  \
ETid                                                          
ETid-914                                    Art. 6 (1) GDPR   
ETid-913                                   Art. 58 (1) GDPR   
ETid-912                                       Art. 13 GDPR   
ETid-911                                       Art. 13 GDPR   
ETid-910  Art. 5 (1) c) GDPR, Art. 5 (1) e) GDPR, Art. 5...   
...                                                     ...   
ETid-5                      Art. 5 (1) b) GDPR, Art. 6 GDPR   
ETid-4    Art. 5 (1) a) GDPR, Art. 5 (1) c) GDPR, Art. 6...   
ETid-3                      Art. 5 (1) a) GDPR, Art. 6 GDPR   
ETid-2              Art. 5 GDPR, Art. 13 GDPR, Art. 14 GDPR   
ETid-1                                         Art. 13 GDPR   

                                                       Type  
ETid                                                         
ETid-914       Insufficient legal basis for data processing  
ETid-913  Insufficient cooperation with supervisory auth...  
ETid-912  Insufficient fulfilment of information obligat...  
ETid-911  Insufficient fulfilment of information obligat...  
ETid-910  Non-compliance with general data processing pr...  
...                                                     ...  
ETid-5         Insufficient legal basis for data processing  
ETid-4         Insufficient legal basis for data processing  
ETid-3         Insufficient legal basis for data processing  
ETid-2         Insufficient legal basis for data processing  
ETid-1    Insufficient fulfilment of information obligat...  

[914 rows x 6 columns]

In [ ]:
df['Authority'] = ""
df['Sector'] = ""
df['Summary'] = ""

In [ ]:
df.loc['ETid-5',:]

Country                                                      BELGIUM
Date of Decision                                          2019-05-28
Fine [€]                                                      2000.0
Controller/Processor                                           Mayor
Quoted Art.                          Art. 5 (1) b) GDPR, Art. 6 GDPR
Type                    Insufficient legal basis for data processing
Summary                                                             
Authority                                                           
Sector                                                              
Name: ETid-5, dtype: object

In [ ]:
df.loc['ETid-5','Summary'] = "Test"
df.loc['ETid-5',:]

Country                                                      BELGIUM
Date of Decision                                          2019-05-28
Fine [€]                                                      2000.0
Controller/Processor                                           Mayor
Quoted Art.                          Art. 5 (1) b) GDPR, Art. 6 GDPR
Type                    Insufficient legal basis for data processing
Summary                                                         Test
Name: ETid-5, dtype: object

## Parsing single case/fine details

**Attention:** This process takes a few hours, because 914 pages have to be opened and parsed in sequence.

In [ ]:
# Import required module
import time
from selenium import webdriver
  
# Import Select class
from selenium.webdriver.support.ui import Select

In [ ]:
df_length = len(df)

In [ ]:
#df_length = 3

# For each entry in the dataframe open the respective case website and parse the details
for i in range(0, df_length):
    ETid_no = i+1
    driver = webdriver.Edge('../srcs/msedgedriver.exe')
    try:
        driver.get(f'https://www.enforcementtracker.com/ETid-{ETid_no}')
        time.sleep(3)
    except:
        print("Opening source failed. Trying it again.")
        driver.get(f'https://www.enforcementtracker.com/ETid-{ETid_no}')
        time.sleep(3)  
    
    table = driver.find_element_by_xpath('//*[@id="penalties"]/tbody/tr[1]')
    table_html = table.get_attribute('innerHTML')
    
    split_text = table_html.split(">")
    split_text = [text for text in split_text if text.startswith('<') == False]
    split_text
    
    authority = split_text[2][:-4]
    sector = split_text[6][:-4]
    summary = split_text[9][:-4]
    
    print(ETid_no)
    print(authority)
    print(sector)
    print(summary)

    df.loc[f'ETid-{ETid_no}','Authority'] = authority
    df.loc[f'ETid-{ETid_no}','Sector'] = sector
    df.loc[f'ETid-{ETid_no}','Summary'] = summary

    driver.close()


451
Romanian National Supervisory Authority for Personal Data Processing (ANSPDCP)
Media, Telecoms and Broadcasting
The Romanian DPA (ANSPDCP) imposed a fine in the amount of EUR 4,000 on Vodafone România SA. The fine was imposed as a result of complaints alleging that the operator failed to respond to requests for access and erasure of data.
The operator could not provide any evidence for exonaration.
452
Spanish Data Protection Authority (aepd)
Industry and Commerce
The Spanish DPA (AEPD) imposed a fine on Anmavas 61, S.L. for neither granting nor justifiably denying the right to erasure to the data subject, even after receiving a warning issued by the AEPD.
453
Romanian National Supervisory Authority for Personal Data Processing (ANSPDCP)
Industry and Commerce
Due to inadequate technical and organizational measures, the company disclosed the order, delivery and personal data of over 1000 customers via its web store.  The data was displayed on a document in the web store that could b

In [ ]:
df

Country Date of Decision Fine [€]  \
ETid                                          
ETid-914    SPAIN       2021-11-23    40000   
ETid-913    SPAIN       2021-11-23     3000   
ETid-912    SPAIN       2021-11-22     2000   
ETid-911    SPAIN       2021-11-22     3000   
ETid-910   FRANCE       2021-11-04   400000   
...           ...              ...      ...   
ETid-5    BELGIUM       2019-05-28   2000.0   
ETid-4    AUSTRIA       2018-12-20   2200.0   
ETid-3    AUSTRIA       2018-09-27    300.0   
ETid-2    AUSTRIA             2018   1800.0   
ETid-1    AUSTRIA       2018-12-09   4800.0   

                             Controller/Processor  \
ETid                                                
ETid-914                     Vodafone España, SAU   
ETid-913                           FUENSANTA S.L.   
ETid-912               ANIVERSALIA NETWORKS, S.L.   
ETid-911                                  Unknown   
ETid-910  Régie autonome des transports parisiens   
...                                           ...   
ETid-5                                      Mayor   
ETid-4                             Private person   
ETid-3                          Private car owner   
ETid-2                           Kebab restaurant   
ETid-1                              Betting place   

                                                Quoted Art.  \
ETid                                                          
ETid-914                                    Art. 6 (1) GDPR   
ETid-913                                   Art. 58 (1) GDPR   
ETid-912                                       Art. 13 GDPR   
ETid-911                                       Art. 13 GDPR   
ETid-910  Art. 5 (1) c) GDPR, Art. 5 (1) e) GDPR, Art. 5...   
...                                                     ...   
ETid-5                      Art. 5 (1) b) GDPR, Art. 6 GDPR   
ETid-4    Art. 5 (1) a) GDPR, Art. 5 (1) c) GDPR, Art. 6...   
ETid-3                      Art. 5 (1) a) GDPR, Art. 6 GDPR   
ETid-2              Art. 5 GDPR, Art. 13 GDPR, Art. 14 GDPR   
ETid-1                                         Art. 13 GDPR   

                                                       Type  \
ETid                                                          
ETid-914       Insufficient legal basis for data processing   
ETid-913  Insufficient cooperation with supervisory auth...   
ETid-912  Insufficient fulfilment of information obligat...   
ETid-911  Insufficient fulfilment of information obligat...   
ETid-910  Non-compliance with general data processing pr...   
...                                                     ...   
ETid-5         Insufficient legal basis for data processing   
ETid-4         Insufficient legal basis for data processing   
ETid-3         Insufficient legal basis for data processing   
ETid-2         Insufficient legal basis for data processing   
ETid-1    Insufficient fulfilment of information obligat...   

                                                    Summary  \
ETid                                                          
ETid-914  The Spanish DPA (AEPD) has fined Vodafone Espa...   
ETid-913  The controller failed to provide information r...   
ETid-912  The Spanish DPA (AEPD) fined ANIVERSALIA NETWO...   
ETid-911  The Spanish DPA (AEPD) has imposed a fine of E...   
ETid-910  The French DPA (CNIL) imposed a fine of EUR 40...   
...                                                     ...   
ETid-5    The administrative fine was imposed for the mi...   
ETid-4    The fine was imposed against a private person ...   
ETid-3                       A Dashcam was unlawfully used.   
ETid-2    CCTV was unlawfully used. Sufficient informati...   
ETid-1    Video surveillance was not sufficiently marked...   

                                         Authority  \
ETid                                                 
ETid-914  Spanish Data Protection Authority (aepd)   
ETid-913  Spanish Data Protection Authority (aepd)   
ETid-912  Spanish Data Protectio

In [ ]:
df.reset_index(inplace=True)
df

In [ ]:
# Using pickle format to preserve formats
pd.to_pickle(df,"../data/external/gdpr_fines_enforcementtracker.pkl", protocol=2)

## Checking single case entries for errors

- One error for case id **907**, because not sector was assigned

In [ ]:
df['Sector'].unique()

array([nan, 'Industry and Commerce', 'Not assigned',
       'Transportation and Energy', 'Media, Telecoms and Broadcasting',
       'Finance, Insurance and Consulting',
       'Individuals and Private Associations',
       'Public Sector and Education', 'Real Estate',
       'Accomodation and Hospitalty', 'Health Care', 'Employment'],
      dtype=object)

In [ ]:
df[df['Sector'] == 'Art. 13 GDPR']

ETid Country Date of Decision Fine [€]    Controller/Processor  \
7  ETid-907   SPAIN       2021-11-22     1000  Neighborhood community   

    Quoted Art.                                               Type Summary  \
7  Art. 13 GDPR  Insufficient fulfilment of information obligat...     lin   

                                  Authority        Sector  
7  Spanish Data Protection Authority (aepd)  Art. 13 GDPR

In [ ]:
df['Authority'].unique()

array(['Spanish Data Protection Authority (aepd)',
       'French Data Protection Authority (CNIL)',
       'Cypriot Data Protection Commissioner',
       'Bulgarian Commission for Personal Data Protection (KZLD)',
       'Dutch Supervisory Authority for Data Protection (AP)',
       'Polish National Personal Data Protection Office (UODO)',
       'Romanian National Supervisory Authority for Personal Data Processing (ANSPDCP)',
       'Hellenic Data Protection Authority (HDPA)',
       'National Commission for Data Protection (CNPD)',
       'Data Protection Authority of Ireland',
       'Italian Data Protection Authority (Garante)',
       'Information Commissioner (ICO)',
       'Norwegian Supervisory Authority (Datatilsynet)',
       'Austrian Data Protection Authority (dsb)',
       'Danish Data Protection Authority (Datatilsynet)',
       'Information Commissioner of Isle of Man',
       'Data Protection Authority of Hamburg',
       'Hungarian National Authority for Data Protecti

In [ ]:
df.set_index('ETid', inplace=True)
df

Country Date of Decision Fine [€]  \
ETid                                          
ETid-914    SPAIN       2021-11-23    40000   
ETid-913    SPAIN       2021-11-23     3000   
ETid-912    SPAIN       2021-11-22     2000   
ETid-911    SPAIN       2021-11-22     3000   
ETid-910   FRANCE       2021-11-04   400000   
...           ...              ...      ...   
ETid-5    BELGIUM       2019-05-28   2000.0   
ETid-4    AUSTRIA       2018-12-20   2200.0   
ETid-3    AUSTRIA       2018-09-27    300.0   
ETid-2    AUSTRIA             2018   1800.0   
ETid-1    AUSTRIA       2018-12-09   4800.0   

                             Controller/Processor  \
ETid                                                
ETid-914                     Vodafone España, SAU   
ETid-913                           FUENSANTA S.L.   
ETid-912               ANIVERSALIA NETWORKS, S.L.   
ETid-911                                  Unknown   
ETid-910  Régie autonome des transports parisiens   
...                                           ...   
ETid-5                                      Mayor   
ETid-4                             Private person   
ETid-3                          Private car owner   
ETid-2                           Kebab restaurant   
ETid-1                              Betting place   

                                                Quoted Art.  \
ETid                                                          
ETid-914                                    Art. 6 (1) GDPR   
ETid-913                                   Art. 58 (1) GDPR   
ETid-912                                       Art. 13 GDPR   
ETid-911                                       Art. 13 GDPR   
ETid-910  Art. 5 (1) c) GDPR, Art. 5 (1) e) GDPR, Art. 5...   
...                                                     ...   
ETid-5                      Art. 5 (1) b) GDPR, Art. 6 GDPR   
ETid-4    Art. 5 (1) a) GDPR, Art. 5 (1) c) GDPR, Art. 6...   
ETid-3                      Art. 5 (1) a) GDPR, Art. 6 GDPR   
ETid-2              Art. 5 GDPR, Art. 13 GDPR, Art. 14 GDPR   
ETid-1                                         Art. 13 GDPR   

                                                       Type  \
ETid                                                          
ETid-914       Insufficient legal basis for data processing   
ETid-913  Insufficient cooperation with supervisory auth...   
ETid-912  Insufficient fulfilment of information obligat...   
ETid-911  Insufficient fulfilment of information obligat...   
ETid-910  Non-compliance with general data processing pr...   
...                                                     ...   
ETid-5         Insufficient legal basis for data processing   
ETid-4         Insufficient legal basis for data processing   
ETid-3         Insufficient legal basis for data processing   
ETid-2         Insufficient legal basis for data processing   
ETid-1    Insufficient fulfilment of information obligat...   

                                                    Summary  \
ETid                                                          
ETid-914  The Spanish DPA (AEPD) has fined Vodafone Espa...   
ETid-913  The controller failed to provide information r...   
ETid-912  The Spanish DPA (AEPD) fined ANIVERSALIA NETWO...   
ETid-911  The Spanish DPA (AEPD) has imposed a fine of E...   
ETid-910  The French DPA (CNIL) imposed a fine of EUR 40...   
...                                                     ...   
ETid-5    The administrative fine was imposed for the mi...   
ETid-4    The fine was imposed against a private person ...   
ETid-3                       A Dashcam was unlawfully used.   
ETid-2    CCTV was unlawfully used. Sufficient informati...   
ETid-1    Video surveillance was not sufficiently marked...   

                                         Authority  \
ETid                                                 
ETid-914  Spanish Data Protection Authority (aepd)   
ETid-913  Spanish Data Protection Authority (aepd)   
ETid-912  Spanish Data Protectio

### Fixing ETid 907 (because not sector was assigned)

In [ ]:
ETid_no = 907 # ETid to be fixed
driver = webdriver.Edge('msedgedriver.exe')
driver.get(f'https://www.enforcementtracker.com/ETid-{ETid_no}')
time.sleep(3)
table = driver.find_element_by_xpath('//*[@id="penalties"]/tbody/tr[1]')
table_html = table.get_attribute('innerHTML')

split_text = table_html.split(">")
split_text = [text for text in split_text if text.startswith('<') == False]
split_text


['ETid-907</a',
 'SPAIN</td',
 'Spanish Data Protection Authority (aepd)</td',
 '2021-11-22</td',
 '1,000</td',
 'Neighborhood community</td',
 'Art. 13 GDPR</td',
 'Insufficient fulfilment of information obligations</td',
 'The Spanish DPA has imposed a fine of EUR 1000 on a neighborhood community. The reason for this was that the information sign about a video surveillance system did not contain sufficient information as required by Art. 13 GDPR. The sign contained neither a reference to the data controller nor an  address to contact if one wishes to exercise their data subjects rights.</td',
 'link</a',
 'https://www.enforcementtracker.com/ETid-907</a',
 ' (Copy: <a id="myShareButton" class="myShare907" href="#/" onclick="copyETID(907);return false;" alt="Copy URL to clipboard"',
 ') | Short URL: <a href="https://etid.link/ETid-907" target="_blank"',
 'https://etid.link/ETid-907</a',
 '']

In [ ]:
authority = split_text[2][:-4]
sector = 'Not assigned'
summary = split_text[8][:-4]

print(ETid_no)
print(authority)
print(sector)
print(summary)


907
Spanish Data Protection Authority (aepd)
Not assigned
The Spanish DPA has imposed a fine of EUR 1000 on a neighborhood community. The reason for this was that the information sign about a video surveillance system did not contain sufficient information as required by Art. 13 GDPR. The sign contained neither a reference to the data controller nor an  address to contact if one wishes to exercise their data subjects rights.


In [ ]:
df.loc[f'ETid-{ETid_no}','Authority'] = authority
df.loc[f'ETid-{ETid_no}','Sector'] = sector
df.loc[f'ETid-{ETid_no}','Summary'] = summary
driver.close()


In [ ]:
df.reset_index(inplace=True)
df

ETid  Country Date of Decision Fine [€]  \
0    ETid-914    SPAIN       2021-11-23    40000   
1    ETid-913    SPAIN       2021-11-23     3000   
2    ETid-912    SPAIN       2021-11-22     2000   
3    ETid-911    SPAIN       2021-11-22     3000   
4    ETid-910   FRANCE       2021-11-04   400000   
..        ...      ...              ...      ...   
909    ETid-5  BELGIUM       2019-05-28   2000.0   
910    ETid-4  AUSTRIA       2018-12-20   2200.0   
911    ETid-3  AUSTRIA       2018-09-27    300.0   
912    ETid-2  AUSTRIA             2018   1800.0   
913    ETid-1  AUSTRIA       2018-12-09   4800.0   

                        Controller/Processor  \
0                       Vodafone España, SAU   
1                             FUENSANTA S.L.   
2                 ANIVERSALIA NETWORKS, S.L.   
3                                    Unknown   
4    Régie autonome des transports parisiens   
..                                       ...   
909                                    Mayor   
910                           Private person   
911                        Private car owner   
912                         Kebab restaurant   
913                            Betting place   

                                           Quoted Art.  \
0                                      Art. 6 (1) GDPR   
1                                     Art. 58 (1) GDPR   
2                                         Art. 13 GDPR   
3                                         Art. 13 GDPR   
4    Art. 5 (1) c) GDPR, Art. 5 (1) e) GDPR, Art. 5...   
..                                                 ...   
909                    Art. 5 (1) b) GDPR, Art. 6 GDPR   
910  Art. 5 (1) a) GDPR, Art. 5 (1) c) GDPR, Art. 6...   
911                    Art. 5 (1) a) GDPR, Art. 6 GDPR   
912            Art. 5 GDPR, Art. 13 GDPR, Art. 14 GDPR   
913                                       Art. 13 GDPR   

                                                  Type  \
0         Insufficient legal basis for data processing   
1    Insufficient cooperation with supervisory auth...   
2    Insufficient fulfilment of information obligat...   
3    Insufficient fulfilment of information obligat...   
4    Non-compliance with general data processing pr...   
..                                                 ...   
909       Insufficient legal basis for data processing   
910       Insufficient legal basis for data processing   
911       Insufficient legal basis for data processing   
912       Insufficient legal basis for data processing   
913  Insufficient fulfilment of information obligat...   

                                               Summary  \
0    The Spanish DPA (AEPD) has fined Vodafone Espa...   
1    The controller failed to provide information r...   
2    The Spanish DPA (AEPD) fined ANIVERSALIA NETWO...   
3    The Spanish DPA (AEPD) has imposed a fine of E...   
4    The French DPA (CNIL) imposed a fine of EUR 40...   
..                                                 ...   
909  The administrative fine was imposed for the mi...   
910  The fine was imposed against a private person ...   
911                     A Dashcam was unlawfully used.   
912  CCTV was unlawfully used. Sufficient informati...   
913  Video surveillance was not sufficiently marked...   

                                    Authority  \
0    Spanish Data Protection Authority (aepd)   
1    Spanish Data Protection Authority (aepd)   
2    Spanish Data Protection Authority (aepd)   
3    Spanish Data Protection Authority (aepd)   
4     French Data Protection Authority (CNIL)   
..                                        ...   
909   Belgian Data Protection Authority (APD)   
910  Austrian Data Protection Authority (dsb)   
911  Austrian Data Protection Authority (dsb)   
912  Austrian Data Protection Authority (dsb)   
913  Austrian Data Protection Authority (dsb)   

                                   Sector  
0        Media, Telecoms and Broadcasting  
1                   Industry and Commer

In [ ]:
# Using pickle format to preserve formats
pd.to_pickle(df,"../data/external/gdpr_fines_enforcementtracker.pkl",protocol=2)